data-collection/data-collection/View Controllers/Floating Panel View Controller/FloatingPanelViewController.swift

```diff
@@ -0,0 +1,484 @@
+    private var internalEdgeInsets: UIEdgeInsets {
+        return isCompactWidth ? compactWidthInsets : regularWidthInsets
+    }
+    
+    /// Denotes whether the floating panel view controller is being displayed in
+    /// a compact width layout.
+    private var isCompactWidth: Bool = false {
+        didSet {
+            // Enable the correct handlebar (top vs. bottom) for the current layout.
+            updateHandlebarVisibility()
+            updateInterfaceForCurrentTraits()
+            
+            // Set corner masks
+            view.layer.maskedCorners = isCompactWidth ?
+                [.layerMinXMinYCorner, .layerMaxXMinYCorner] :
+                [.layerMinXMinYCorner, .layerMaxXMinYCorner, .layerMaxXMaxYCorner, .layerMinXMaxYCorner]
+        }
+    }
+    
+    /// Determines whether we've initialized the initial layout constraints.
+    private var initialized = false
+    
+    /// Constraints for the regular width size class.
+    private var regularWidthConstraints = [NSLayoutConstraint]()
+    
+    /// Constraints for the compact width size class.
+    private var compactWidthConstraints = [NSLayoutConstraint]()
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        
+        isCompactWidth = traitCollection.horizontalSizeClass == .compact
+        
+        // Initialize controls from properties that may have been set before
+        // the view was loaded.
+        updateImage()
+        updateCloseButton()
+        updateHeaderStackView()
+        updateAllowManualResize()
+        updateFloatingPanelTitle()
+        updateFloatingPanelSubtitle()
+    }
+    
+    /// Sets up the constrains for both regular and compact horizontal size classes.
+    /// - Parameter superview: The superview of the floating panel.
+    private func setupConstraints() {
+        guard let superview = view.superview else { return }
+        
+        // Set the resizable constraint used by the panGestureRecognizer to resize the panel.
+        resizeableLayoutConstraint = view.heightAnchor.constraint(equalToConstant: 0)
+        resizeableLayoutConstraint.priority = .defaultLow
+        
+        // updateState will update the resizableLayoutConstraint constant
+        // based on the current state.
+        updateState(false)
+        
+        // Define the constraints used for a regular-width layout.
+        regularWidthConstraints = [
+            view.leadingAnchor.constraint(equalTo: superview.safeAreaLayoutGuide.leadingAnchor, constant: regularWidthInsets.left),
+            view.widthAnchor.constraint(equalToConstant: floatingPanelWidth),
+            view.topAnchor.constraint(equalTo: superview.safeAreaLayoutGuide.topAnchor, constant: regularWidthInsets.top),
+            view.bottomAnchor.constraint(lessThanOrEqualTo: superview.safeAreaLayoutGuide.bottomAnchor, constant: -regularWidthInsets.bottom),
```

Comment:
I personally find negative constants a little harder to reason about, so I avoid them whenever possible by putting the trailing-most (or bottom-most) view on the left hand side of the constraint equation. For example, `view.bottomAnchor` and `superview.safeAreaLayoutGuide.bottomAnchor` could be swapped to make the constant positive:

```suggestion
            superview.safeAreaLayoutGuide.bottomAnchor.constraint(greaterThanOrEqualTo: view.bottomAnchor, constant: regularWidthInsets.bottom),
```

But that's just a preference. You don't have to do it that way.

Reply:
That works for me.